tests/components/sunweg/test_config_flow.py

```diff
@@ -0,0 +1,135 @@
+"""Tests for the Sun WEG server config flow."""
+from copy import deepcopy
+from unittest.mock import patch
+
+from sunweg.api import APIHelper
+
+from homeassistant import config_entries, data_entry_flow
+from homeassistant.components.sunweg.const import CONF_PLANT_ID, DOMAIN
+from homeassistant.const import CONF_PASSWORD, CONF_USERNAME
+from homeassistant.core import HomeAssistant
+
+from .common import FIXTURE_USER_INPUT, SUNWEG_LOGIN_RESPONSE, SUNWEG_PLANT_RESPONSE
+
+from tests.common import MockConfigEntry
+
+
+async def test_show_authenticate_form(hass: HomeAssistant) -> None:
+    """Test that the setup form is served."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+
+    assert result["type"] == data_entry_flow.FlowResultType.FORM
+    assert result["step_id"] == "user"
+
+
+async def test_incorrect_login(hass: HomeAssistant) -> None:
+    """Test that it shows the appropriate error when an incorrect username/password/server is entered."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+
+    with patch.object(APIHelper, "authenticate", return_value=False):
+        result = await hass.config_entries.flow.async_configure(
+            result["flow_id"], FIXTURE_USER_INPUT
+        )
+
+    assert result["type"] == data_entry_flow.FlowResultType.FORM
+    assert result["step_id"] == "user"
+    assert result["errors"] == {"base": "invalid_auth"}
+
+
+async def test_no_plants_on_account(hass: HomeAssistant) -> None:
+    """Test registering an integration and finishing flow with an entered plant_id."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+    user_input = FIXTURE_USER_INPUT.copy()
+
+    with patch.object(
+        APIHelper, "authenticate", return_value=SUNWEG_LOGIN_RESPONSE
```

Comment:
this one screams "convert it to a fixture" to me because it's used multiple times